### PR TITLE
add satellite main structure

### PR DESCRIPTION
**Pull request by EvanLjp**
add satellite main structure, would add some unit test later.

### CODE REVIEW

**Comment by gxthrj** [comment]
Huge PR

**Comment by EvanLjp** [comment]
> Huge PR

would add some doc later

**Comment by wu-sheng** [comment]
I will wait for the doc.

**Comment by EvanLjp** [comment]
@wu-sheng some core design doc is pushed and each specific module has comments. If anything is not clear, please let me know.

**Comment by nic-chen** [comment]
Maybe we could split it into several small PRs and merge them one by one?

**Comment by EvanLjp** [comment]
> Maybe we could split it into several small PRs and merge them one by one?

no problem

**Comment by EvanLjp** [comment]
@nic-chen already create a new PR. And please don't delete the current branch before the small PRs are approved.

**Comment by wu-sheng** [comment]
I just asked @kezhenxu94 to open required CIs and auto-update.

**Comment by wu-sheng** [comment]
What is the relationship between this and #8 ?

**Comment by EvanLjp** [comment]
> What is the relationship between this and #8 ?

@nic-chen thinks the pr is too huge, suggest me to split it.

File: internal/satellite/boot/boot.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+	"os/signal"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/log"
+	"github.com/apache/skywalking-satellite/internal/satellite/config"
+	"github.com/apache/skywalking-satellite/internal/satellite/module"
+)
+
+// Start Satellite.
+func Start(cfg *config.SatelliteConfig) error {
```
Comment (surlymo) [inline internal/satellite/boot/boot.go:34]: it seems simpler if merging defaultplugin & service design. plugin only have four layers:
plugin basic init -> lifecycle management -> specific plugin interface -> specific plugin implements

**Pull request closed without merging**